feat: add C implementation `stats/base/dists/beta/mgf`

ShabiShett07 opened the pull request:
Resolves #3430 .

## Description

> What is the purpose of this pull request?

This pull request:

-  Solves the issue #3430

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3430 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add C implementation /stats/base/dists/beta/mgf

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

ShabiShett07 commented:
/stdlib lint-autofix

ShabiShett07 committed:
chore: update implementation

ShabiShett07 committed:
Merge branch 'develop' into feature/beta/mgf

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore:  add indentation

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/lib/native.js:
```diff
@@ -0,0 +1,87 @@
+
+/**
+* Evaluates the moment-generating function (MGF) for a beta distribution with first shape parameter `alpha` and second shape parameter `beta` at a value `t`.
+*
+* @param {number} t - input value
```
```suggestion
*
* @private
* @param {number} t - input value
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/README.md:
```diff
@@ -166,6 +166,104 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_beta_mgf( t, alpha, beta )
+
+Returns the [mgf][mgf] of a [beta][beta-distribution] distribution with parameters `alpha` (first shape parameter) and `beta` (second shape parameter) with input value `t`.
```
```suggestion
Evaluates the [moment-generating function][mgf] (MGF) for a [beta][beta-distribution] distribution with parameters `alpha` (first shape parameter) and `beta` (second shape parameter).
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/README.md:
```diff
@@ -166,6 +166,104 @@ for ( i = 0; i < 10; i++ ) {
+    double alpha;
+    double beta;
+    double y;
+    int i;
```
```suggestion
    double alpha;
    double beta;
    double t;
    double y;
    int i;
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/README.md:
```diff
@@ -166,6 +166,104 @@ for ( i = 0; i < 10; i++ ) {
+        alpha = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
+        beta = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
+        y = stdlib_base_dists_beta_mgf( alpha, beta );
+        printf( "t: %lf, alpha: %lf, beta: %lf, mgf(X;t,alpha,beta): %lf\n", t alpha, beta, y );
```
```suggestion
        printf( "t: %lf, α: %lf, β: %lf, M_X(t;α,β): %lf\n", t, alpha, beta, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/examples/c/example.c:
```diff
@@ -0,0 +1,43 @@
+		alpha = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
+		beta = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
+		y = stdlib_base_dists_beta_mgf( t, alpha, beta );
+		printf( "t: %lf, alpha: %lf, beta: %lf, mgf(X;t,alpha,beta): %lf\n", t, alpha, beta, y );
```
```suggestion
		printf( "t: %lf, α: %lf, β: %lf, M_X(t;α,β): %lf\n", t, alpha, beta, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/include/stdlib/stats/base/dists/beta/mgf.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Returns the mgf of a beta distribution.
```
```suggestion
* Evaluates the moment-generating function (MGF) for a beta distribution with first shape parameter `alpha` and second shape parameter `beta` at a value `t`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/src/main.c:
```diff
@@ -0,0 +1,70 @@
+* limitations under the License.
+*/
+
+// MODULES //
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/beta.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/constants/float64/eps.h"
+
+
+// MAIN //
+
+/**
+* Evaluates the moment-generating function (MGF) for a beta distribution with first shape parameter `alpha` and second shape parameter `beta` at a value `t`
+*
+* @param t      input value
+* @param alpha  first shape parameter
+* @param beta   second shape parameter
+* @returns      evaluated MGF
+*
+* @example
+* double y = stdlib_base_dists_beta_mgf( 0.5, 1.0, 1.0 );
+* // returns ~1.297
+*/
```
```suggestion
#include "stdlib/math/base/assert/is_nan.h"
#include "stdlib/math/base/special/beta.h"
#include "stdlib/math/base/special/abs.h"
#include "stdlib/constants/float64/eps.h"

/**
* Evaluates the moment-generating function (MGF) for a beta distribution with first shape parameter `alpha` and second shape parameter `beta` at a value `t`
*
* @param t      input value
* @param alpha  first shape parameter
* @param beta   second shape parameter
* @return       evaluated MGF
*
* @example
* double y = stdlib_base_dists_beta_mgf( 0.5, 1.0, 1.0 );
* // returns ~1.297
*/
```

ShabiShett07 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/beta/mgf

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: update variable order

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

Planeshifter committed:
fix: modernize PRNG usage and fix C variable order in beta/mgf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/src/main.c:
```diff
@@ -0,0 +1,71 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// MODULES //
+
+#include "stdlib/stats/base/dists/beta/mgf.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/beta.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/constants/float64/eps.h"
+
+
+// MAIN //
+
+/**
```
```suggestion
*/

#include "stdlib/stats/base/dists/beta/mgf.h"
#include "stdlib/math/base/assert/is_nan.h"
#include "stdlib/math/base/special/beta.h"
#include "stdlib/math/base/special/abs.h"
#include "stdlib/constants/float64/eps.h"

/**
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/beta/mgf/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/stats/base/dists/beta/mgf.h"
+#include "stdlib/math/base/napi/ternary.h"
+
+// cppcheck-suppress shadowFunction
```
```suggestion
```

Planeshifter committed:
chore: remove cppcheck directive and heading comments

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Should be ready now. Thanks, @ShabiShett07!
